src/tests/unit/platform/test_alertsv7_api.py

```diff
@@ -568,6 +568,121 @@ def on_post(url, body, **kwargs):
+                                     "threat_id": ["B0RG"], "type": ["INTRUSION_DETECTION_SYSTEM"],
+                                     "policy_applied": ["APPLIED"],
+                                     "reason_code": ["ATTACK_VECTOR"], "run_state": ["RAN"], "sensor_action": ["DENY"],
+                                     "alert_notes_present": ["True"], "attack_tactic": ["tactic"],
+                                     "attack_technique": ["technique"], "blocked_effective_reputation": ["NOT_LISTED"],
+                                     "blocked_md5": ["md5_hash"], "blocked_name": ["tim"],
+                                     "blocked_sha256": ["sha256_hash"],
+                                     "ttps": ["malicious"], "tms_rule_id": ["xtvr"],
+                                     "workflow_status": ["OPEN"],
+                                     "netconn_remote_ip": ["10.29.99.1"], "netconn_remote_domain": ["example.com"],
+                                     "netconn_protocol": ["TCP"], "netconn_remote_port": ["54321"],
+                                     "netconn_local_port": ["12345"],
+                                     "childproc_cmdline": ["/usr/bin/python"],
+                                     "childproc_effective_reputation": ["PUP"],
+                                     "childproc_guid": ["12345678"], "childproc_name": ["python"],
+                                     "childproc_sha256": ["sha256_child"], "childproc_username": ["steven"],
+                                     "primary_event_id": ["123456"],
+                                     "parent_cmdline": ["/usr/bin/python"],
+                                     "parent_effective_reputation": ["PUP"],
+                                     "parent_guid": ["12345678"], "parent_name": ["python"],
+                                     "parent_sha256": ["sha256_parent"], "parent_username": ["steven"],
+                                     "process_cmdline": ["/usr/bin/python"],
+                                     "process_effective_reputation": ["PUP"],
+                                     "process_guid": ["12345678"], "process_name": ["IEXPLORE.EXE"],
+                                     "process_sha256": ["0123456789ABCDEF0123456789ABCDEF"],
+                                     "process_username": ["steven"],
+                                     "process_reputation": ["SUSPECT_MALWARE"],
+                                     "process_issuer": ["Microsoft"], "process_publisher": ["Microsoft"]},
+                        "sort": [{"field": "name", "order": "DESC"}]}
+        return {"results": [{"id": "S0L0", "org_key": "test", "threat_id": "B0RG",
+                             "workflow": {"status": "OPEN"}}], "num_found": 1}
+
+    cbcsdk_mock.mock_request('POST', "/api/alerts/v7/orgs/test/alerts/_search", on_post)
+    api = cbcsdk_mock.api
+
+    query = api.select(Alert).where("Blort").add_criteria("type", ["INTRUSION_DETECTION_SYSTEM"])\
```

Comment:
Two additional tests - instantiate class, and class as string - select("IntrusionDetectionSystemAlert")